Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -1365,7 +1368,8 @@ def visit_WithStatNode(self, node):
                     name, value = directive
                     if name in ('nogil', 'gil'):
                         # special case: in pure mode, "with nogil" spells "with cython.nogil"
-                        node = Nodes.GILStatNode(node.pos, state = name, body = node.body)
+                        condition = value if isinstance(value, ExprNodes.PrimaryCmpNode) else None
```

Comment:
`else None`? But then we'd ignore existing user code. I think we should always honour what's there, as we do with the Cython syntax. Ideally, both should be handled in the same place, or failing that, at least in the same way.

Reply:
Fixed